var/spack/repos/builtin/packages/python/package.py

```diff
@@ -114,22 +115,28 @@ def install(self, spec, prefix):
                       'user configurations are present.').format(self.version))
 
         # Need this to allow python build to find the Python installation.
-        env['PYTHONHOME'], env['PYTHONPATH'] = prefix, prefix
-        env['MACOSX_DEPLOYMENT_TARGET'] = '10.6'
+        spack_env.set('PYTHONHOME', prefix)
+        spack_env.set('PYTHONPATH', prefix)
+        spack_env.set('MACOSX_DEPLOYMENT_TARGET', platform.mac_ver()[0])
+
+    def configure_args(self):
+        spec = self.spec
 
-        # Rest of install is pretty standard except setup.py needs to
-        # be able to read the CPPFLAGS and LDFLAGS as it scans for the
-        # library and headers to build
+        # setup.py needs to be able to read the CPPFLAGS and LDFLAGS
+        # as it scans for the library and headers to build
         dep_pfxs = [dspec.prefix for dspec in spec.dependencies('link')]
         config_args = [
-            '--prefix={0}'.format(prefix),
             '--with-threads',
-            '--enable-shared',
             'CPPFLAGS=-I{0}'.format(' -I'.join(dp.include for dp in dep_pfxs)),
             'LDFLAGS=-L{0}'.format(' -L'.join(dp.lib for dp in dep_pfxs)),
         ]
-        if spec.satisfies("platform=darwin") and ('%gcc' in spec):
-            config_args.append('--disable-toolbox-glue')
+        if spec.satisfies('platform=darwin'):
+            if '%gcc' in spec:
+                config_args.append('--disable-toolbox-glue')
+        else:
+            # --enable-shared causes problems on macOS and is not recommended:
+            # http://bugs.python.org/issue29846
+            config_args.append('--enable-shared')
```

Comment:
i see that you

> happen to have both Python 2 and 3 installed with both Homebrew and Anaconda

so maybe that's somehow breaks things. I only have native python form osx and the one installed by Spack.